main.rb

```diff
@@ -0,0 +1,119 @@
+require './app'
+
+class Main
+  def initialize(message, app)
+    @message = message
+    @app = app
+  end
+
+  def check_permission_input(permission_input)
+    case permission_input
+    when 'Y'
+      true
+    when 'N'
+      false
+    else
+      puts 'Please Enter a Valid option'
+      exit
+    end
+  end
+
+  def create_person_by_type(age_input, name_input, type_input)
+    case type_input
+    when '1'
+      print "Does the student have their parent's permission? [Y/N] "
+      permission_input = gets.chomp.upcase
+      permission = check_permission_input(permission_input)
+      @app.create_person(age_input, name_input, type_input, permission)
+    when '2'
+      print 'What is the specialization of the teacher? '
+      spec_input = gets.chomp
+      @app.create_person(age_input, name_input, type_input, nil, spec_input)
+    else
+      puts 'Please Enter a Valid option'
```

Comment:
- [Optional] @TechnoAndy I think validation here is performed late. As you can see in the screenshot below, I could still proceed to create a student(name, age) or teacher(name, age) after entering an invalid input(**r**). It will be better if the validation is done immediately when the wrong input is entered

**_Screenshot_**
![Capture](https://user-images.githubusercontent.com/63087491/216628671-54c7e939-def0-4c16-91ee-80ff40f0e6c8.PNG)